packages/mui-system/src/createBox.d.ts

```diff
@@ -1,7 +1,12 @@
-import Box from './Box';
+import { OverridableComponent } from '@mui/types';
+import { BoxTypeMap } from './Box';
+import { Theme as SystemTheme } from './createTheme';
 
-export default function createBox(options?: {
-  defaultTheme: object;
+export default function createBox<
+  T extends object = SystemTheme,
+  AdditionalProps extends {} = {},
```

Comment:
if `AdditionalProps` needs to be a plain object it should have `extends Record<string, unknown>`. `extends {}` only guarantees it won't be `null` or `undefined`.